Scope baggage/context interaction functionality (#1525)
open-telemetry/opentelemetry-specification: specification/baggage/api.md

The functions described here are one way to approach interacting with the
`Baggage` via having struct/object that represents the entire Baggage content.
Depending on language idioms, a language API MAY implement these functions by

> interacting with the `Baggage` via the `Context` directly.

Reviewer: ```suggestion
interacting with the Baggage via the `Context` directly.
```
(not referring to the `Baggage` type, but to the concept of Baggage)

Author: ah, SGTM :+1:

This section defines all operations within the Baggage API that interact with
the [`Context`](../context/context.md).

> If an implementation of this API does not interact purely via the `Context`, it

Author: I'm in favor of your suggestion, but I was copying language from the Overview section of the specification that stated the option with these words.

Do you think it is worth updating that section to match? I'm okay with this as well. My original intention was to keep the change as small as possible, but if it isn't clarifying things it isn't worth it either.

> If an implementation of this API does not interact purely via the `Context`, it

Reviewer: I just re-read the above, and suggest the following:

1) change L37-38 to 
```
Depending on language idioms, a language API MAY implement these functions by
interacting with the baggage directly via the `Context` object.
```

2) this line:

```suggestion
If an implementation of this API does not operate directly on the `Context`, it
```
